### PR TITLE
Add higher level utilities for PushManager and Example for Pub/Sub

### PR DESCRIPTION
**Pull request by altanozlu**
I have added two new functions for MultiplexedConnection and ConnectionManager, 
I believe this methods will make easier to use RESP3 with PubSub.

### CODE REVIEW

**Review by nihohit — COMMENTED** [review]
Since we have time to play with the API until the first version with this is released, I'd prefer that instead of these slight improvements, we'd make an effort to consider what is the right API to expose to the user.
Do we really need to expose the `replace_sender` function to the user, or can we just have these functions (or similar ones)? Do we really need the internal replacable sender, or can we make the push manager optional and have it come with a single, irreplacable sender?
Should we expose the internal channel to the user, or just some stream abstraction?
We have an API that works, but until the next release is released, it isn't set in stone. Let's use the time to verify that it's the best API we can have.

File: redis/examples/async-pub-sub-resp3.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+use redis::AsyncCommands;
+
+#[tokio::main]
+async fn main() -> redis::RedisResult<()> {
+    // ?resp3=true enables RESP3 for this connection.
+    // Since RESP3 protocol is used there is no need for creating different connections when using PubSub commands.
+    let client = redis::Client::open("redis://127.0.0.1/?resp3=true").unwrap();
```
[the remark this answers is not in the document]
Comment (altanozlu) [reply redis/examples/async-pub-sub-resp3.rs:7]: Maybe ?protocol=3 ?
https://github.com/redis/redis-specifications/pull/12/files#diff-ae02d54246b69e914a58b918a84bd36f3cb918528cc2057707ed5223798403d6R101


https://github.com/StackExchange/StackExchange.Redis/issues/2353

[the remark this answers is not in the document]
Comment (nihohit) [reply redis/examples/async-pub-sub-resp3.rs:7]: indeed
https://github.com/redis/redis-specifications/blob/1252427cdbc497f66a7f8550c6b5f2f35367dc92/uri/redis.txt#L106

**Comment by altanozlu** [comment]
> Since we have time to play with the API until the first version with this is released, I'd prefer that instead of these slight improvements, we'd make an effort to consider what is the right API to expose to the user. Do we really need to expose the `replace_sender` function to the user, or can we just have these functions (or similar ones)? Do we really need the internal replacable sender, or can we make the push manager optional and have it come with a single, irreplacable sender? Should we expose the internal channel to the user, or just some stream abstraction? We have an API that works, but until the next release is released, it isn't set in stone. Let's use the time to verify that it's the best API we can have.

I believe we should hide pushmanager from user, and I'd keep replaceable sender for reconnections etc.
I'm also still fan of having distinct channels per subscription :)

**Comment by nihohit** [comment]
@altanozlu can this be closed?

**Pull request closed without merging**